ui/legacy/src/components/cstudio-forms/controls/locale-selector.js

```diff
@@ -20,11 +20,7 @@
   }
 
   function formatLanguageMessage(id) {
-    if (CrafterCMSNext.i18n.messages.languages[id]) {
-      return CrafterCMSNext.i18n.intl.formatMessage(CrafterCMSNext.i18n.messages.languages[id]);
-    } else {
-      return id;
-    }
+    return CrafterCMSNext.i18n.messages.languages[id] ?? id;
```

Comment:
This shouldn't even work. You need to remove the languages from the json translation files too.